app/models/manageiq/providers/ibm_cloud/power_virtual_servers/cloud_manager/auth_key_pair.rb

```diff
@@ -1,13 +1,15 @@
 class ManageIQ::Providers::IbmCloud::PowerVirtualServers::CloudManager::AuthKeyPair < ManageIQ::Providers::CloudManager::AuthKeyPair
+  require "sshkey"
   supports :create
   supports :delete
 
   def self.raw_create_key_pair(ext_management_system, create_options)
     ext_management_system.with_provider_connection(:service => "PCloudTenantsSSHKeysApi") do |api|
       tenant_id = ext_management_system.pcloud_tenant_id(api.api_client)
       ssh_key   = IbmCloudPower::SSHKey.new(
-        :name    => create_options['name'],
-        :ssh_key => create_options['public_key']
+        :name        => create_options['name'],
+        :ssh_key     => create_options['public_key'],
+        :fingerprint => SSHKey.sha1_fingerprint(create_options['public_key'])
```

Comment:
I don't see a `fingerprint` attribute for the `IbmCloudPower::SSHKey` class only name, ssh_key, and creation_date.  I don't think you want to pass the fingerprint in to the `pcloud_tenants_sshkeys_post` call.